packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts

```diff
@@ -1,13 +1,145 @@
+            type: 'final',
+            // compatibility with BTC PrecomposedTransaction from @trezor/connect
+            transaction: {
+                inputs: [],
+                outputsPermutation: [0],
+                outputs: [
+                    {
+                        address: output.address,
+                        amount,
+                        script_type: 'PAYTOADDRESS',
+                    },
+                ],
+            },
+        };
+    }
+
+    return payloadData;
+};
+
+export const composeTransaction =
+    (formValues: FormState, formState: ComposeActionContext) => () => {
+        const { account, network, feeInfo } = formState;
+        const composeOutputs = getExternalComposeOutput(formValues, account, network);
+        if (!composeOutputs) return; // no valid Output
+
+        const { output, decimals } = composeOutputs;
+
+        // TODO(vl): get custom fee limit from TrezorConnect.BlockchainEstimateFee
+        const customFeeLimit: string | undefined = undefined; // FEE_PER_SIGNATURE.toString();
+
+        // FeeLevels are read-only
+        const levels = customFeeLimit ? feeInfo.levels.map(l => ({ ...l })) : feeInfo.levels;
+        const predefinedLevels = levels.filter(l => l.label !== 'custom');
+        // update predefined levels with customFeeLimit (gasLimit from data size or erc20 transfer)
+        if (customFeeLimit) {
+            predefinedLevels.forEach(l => (l.feeLimit = customFeeLimit));
+        }
+        // in case when selectedFee is set to 'custom' construct this FeeLevel from values
+        if (formValues.selectedFee === 'custom') {
+            predefinedLevels.push({
+                label: 'custom',
+                feePerUnit: formValues.feePerUnit,
+                feeLimit: formValues.feeLimit,
+                blocks: -1,
+            });
+        }
+
+        const wrappedResponse: PrecomposedLevels = {};
+        const response = predefinedLevels.map(level =>
+            calculate(account.availableBalance, output, level),
+        );
+        response.forEach((tx, index) => {
+            const feeLabel = predefinedLevels[index].label as FeeLevel['label'];
+            wrappedResponse[feeLabel] = tx;
+        });
+
+        // format max (calculate sends it as satoshi)
+        // update errorMessage values (symbol)
+        Object.keys(wrappedResponse).forEach(key => {
+            const tx = wrappedResponse[key];
+            if (tx.type !== 'error') {
+                tx.max = tx.max ? formatAmount(tx.max, decimals) : undefined;
+                tx.estimatedFeeLimit = customFeeLimit;
+            }
+            if (tx.type === 'error' && tx.error === 'AMOUNT_NOT_ENOUGH_CURRENCY_FEE') {
+                tx.errorMessage = {
+                    id: 'AMOUNT_NOT_ENOUGH_CURRENCY_FEE',
+                    values: { symbol: network.symbol.toUpperCase() },
+                };
+            }
+        });
+
+        return Promise.resolve(wrappedResponse);
+    };
```

Comment:
This section is essentially the same as for ETH and will be expanded upon in phase 2 when we handle adjustable fees.